api/src/pcapi/routes/backoffice_v3/offerers.py

```diff
@@ -542,6 +568,77 @@ def toggle_top_actor(offerer_id: int) -> utils.BackofficeResponse:
     return _redirect_after_offerer_validation_action()
 
 
+def _offerer_batch_action(
+    api_function: typing.Callable,
+    success_message: str,
+    form: typing.Callable,
+) -> utils.BackofficeResponse:
+    try:
+        _form = form()
+        offerer_ids = [int(id) for id in _form.object_ids.data.split(",")]
+    except ValueError:
+        flash("Identifiant(s) invalide(s)")
+        return _redirect_after_offerer_validation_action(400)
+
+    offerers = offerers_models.Offerer.query.filter(offerers_models.Offerer.id.in_(offerer_ids)).all()
+
+    for offerer in offerers:
+        if hasattr(form, "tags"):
+            saved_homologation_tags = set(filter_homologation_tags(offerer.tags))
+            tags_to_add = set(_form.tags.data) - saved_homologation_tags
+            tags_to_remove = saved_homologation_tags - set(_form.tags.data)
+            api_function(
+                offerer, current_user, _form.comment.data, tags_to_add=tags_to_add, tags_to_remove=tags_to_remove
+            )
+        elif (
+            hasattr(form, "comment")
+            and isinstance(api_function, type(offerers_api.reject_offerer))
+            and not isinstance(api_function, type(offerers_api.validate_offerer))
+        ):
+            api_function(offerer, current_user, _form.comment.data)
+        else:
+            api_function(offerer, current_user)
+
+    flash(success_message, "success")
+
+    return _redirect_after_offerer_validation_action()
+
+
+@validation_blueprint.route("/offerer/batch-pending", methods=["POST"])
+def batch_set_offerer_pending() -> utils.BackofficeResponse:
+    return _offerer_batch_action(
+        offerers_api.set_offerer_pending,
+        "Les structures ont été mise en attente avec succès",
+        offerer_forms.BatchCommentAndTagOffererForm,
+    )
+
+
+@validation_blueprint.route("/offerer/batch-reject", methods=["POST"])
+def batch_reject_offerer() -> utils.BackofficeResponse:
+    try:
+        return _offerer_batch_action(
+            offerers_api.reject_offerer,
+            "Les structures sélectionnées ont été rejetées avec succès",
+            offerer_forms.BatchOptionalCommentForm,
+        )
+    except offerers_exceptions.OffererAlreadyRejectedException:
+        flash("Une des structures a déjà été rejetée", "error")
+        return _redirect_after_offerer_validation_action(400)
+
+
+@validation_blueprint.route("/offerer/batch-validate", methods=["POST"])
+def batch_validate_offerer() -> utils.BackofficeResponse:
+    try:
+        return _offerer_batch_action(
+            offerers_api.validate_offerer,
+            "Les structures sélectionnées ont été validées avec succès",
+            offerer_forms.BatchOptionalCommentForm,
```

Comment:
On n'a pas de modale pour un commentaire dans le cas de la validation, donc `BatchOptionalCommentForm` me surprend ici

Reply:
C'est le formulaire le plus réduit qui permet de récupérer les `object_ids` dans les fonctions de traitement en batch. C'est le même formulaire pour les `user_offerer` également.

Reply:
Mais ça ne reste pas cohérent d'utiliser `BatchOptionalCommentForm`, car on ne peut pas mettre de commentaire dans cet _endpoint_ de validation.

Je verrais plutôt les définitions suivantes : 
- `BatchForm` hérite de `FlaskForm` en ajoutant `object_ids`
- `BatchOptionalCommentForm` hériterait de `BatchForm` et `OptionalCommentForm`
- `BatchCommentAndTagOffererForm` hériterait de `BatchForm` et `CommentAndTagOffererForm`

Ainsi, on passerait : 
```suggestion
            offerer_forms.BatchForm,
```
formulaire dans lequel on n'a pas le champ `comment`, ce qui aide pour pouvoir tester : `if hasattr(form, "comment"):`

Reply:
En fait ça continue à me gêner de prendre un formulaire avec commentaire dans le cas de validation, dans laquelle on n'a pas de commentaire. Ou alors on le rend possible dans la fonction d'api.
Si tu veux utiliser le formulaire minimal, il suffit de définir :
```
class BatchEmptyForm(FlaskForm):
    object_ids = wtforms.HiddenField("Identifiants à traiter")
```
et l'utiliser pour la validation ; pas besoin d'un chantier technique pour ça et c'est réglé en quelques minutes